### PR TITLE
[BEAM-7372][BEAM-8371][BEAM-9372] Sunset Python 2 and Python 3.5 support.

### PR DESCRIPTION
**Pull request by tvalentyn**
Prior announcement:  https://lists.apache.org/thread.html/rcfac94de806751a0d526da2b0adcc6c7becd1cd7e314cc9ea40882a6%40%3Cdev.beam.apache.org%3E

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by tvalentyn** [comment]
Run Seed Job

**Comment by tvalentyn** [comment]
Run Portable_Python PreCommit

**Comment by tvalentyn** [comment]
Run Seed Job

**Comment by tvalentyn** [comment]
Run Portable_Python PreCommit

**Comment by TheNeuralBit** [comment]
Run XVR_Direct PostCommit

File: .test-infra/jenkins/README.md [hunk 1]
```diff
@@ -109,7 +109,6 @@ Beam Jenkins overview page: [link](https://ci-beam.apache.org/)
...
 | beam_PerformanceTests_TextIOIT | [cron](https://ci-beam.apache.org/job/beam_PerformanceTests_TextIOIT/), [hdfs_cron](https://ci-beam.apache.org/job/beam_PerformanceTests_TextIOIT_HDFS/) | `Run Java TextIO Performance Test` | [![Build Status](https://ci-beam.apache.org/job/beam_PerformanceTests_TextIOIT/badge/icon)](https://ci-beam.apache.org/job/beam_PerformanceTests_TextIOIT) [![Build Status](https://ci-beam.apache.org/job/beam_PerformanceTests_TextIOIT_HDFS/badge/icon)](https://ci-beam.apache.org/job/beam_PerformanceTests_TextIOIT_HDFS) |
-| beam_PerformanceTests_WordCountIT_Py27 | [cron](https://ci-beam.apache.org/job/beam_PerformanceTests_WordCountIT_Py27/) | `Run Python27 WordCountIT Performance Test` | [![Build Status](https://ci-beam.apache.org/job/beam_PerformanceTests_WordCountIT_Py27/badge/icon)](https://ci-beam.apache.org/job/beam_PerformanceTests_WordCountIT_Py27) |
```
Comment (TheNeuralBit) [inline .test-infra/jenkins/README.md:112]: Could you remove the badges from the README.md and .github/PULL_REQUEST_TEMPLATE.md too?

**Comment by tvalentyn** [comment]
> Run XVR_Direct PostCommit

@TheNeuralBit we'd need to re-run a seed job first 

> 09:25:42 Could not determine the dependencies of task ':sdks:python:test-suites:direct:xlang:validatesCrossLanguageRunnerSetup'.
09:25:42 > Task with path ':sdks:python:container:py2:docker' not found in project ':sdks:python:test-suites:direct:xlang'.
09:25:42

**Comment by tvalentyn** [comment]
Run Seed Job

**Comment by tvalentyn** [comment]
cc: @robinyqiu

File: release/src/main/python-release/python_release_automation_utils.sh [hunk 2]
```diff
@@ -78,22 +78,18 @@ function get_version() {
...
+    if [[ $2 == "python3.6" ]]; then
       BEAM_PYTHON_SDK_WHL="apache_beam-$VERSION*-cp36-cp36m-manylinux1_x86_64.whl"
     elif [[ $2 == "python3.7" ]]; then
       BEAM_PYTHON_SDK_WHL="apache_beam-$VERSION*-cp37-cp37m-manylinux1_x86_64.whl"
     elif [[ $2 == "python3.8" ]]; then
-      BEAM_PYTHON_SDK_WHL="apache_beam-$VERSION*-cp38-cp38m-manylinux1_x86_64.whl"
+      BEAM_PYTHON_SDK_WHL="apache_beam-$VERSION*-cp38-cp38-manylinux1_x86_64.whl"
     elif [[ $2 == "python3.9" ]]; then
-      BEAM_PYTHON_SDK_WHL="apache_beam-$VERSION*-cp39-cp39m-manylinux1_x86_64.whl"
+      BEAM_PYTHON_SDK_WHL="apache_beam-$VERSION*-cp39-cp39-manylinux1_x86_64.whl"
```
Comment (TheNeuralBit) [inline release/src/main/python-release/python_release_automation_utils.sh:92]: Is dropping the `m` here intentional?

File: sdks/python/apache_beam/__init__.py [hunk 3]
```diff
@@ -94,9 +86,8 @@
   pass
 else:
   raise RuntimeError(
-      'The Apache Beam SDK for Python is only supported on Python 2.7 or '
-      'Python 3. It is not supported on Python [' + str(sys.version_info) +
-      '].')
+      'The Apache Beam SDK for Python is only supported on Python 3. '
+      'It is not supported on Python [' + str(sys.version_info) + '].')
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/__init__.py:90]: Should we raise this error (or maybe just the warning above) for <3.6?

**Review by TheNeuralBit — APPROVED** [review]
A few minor questions and suggestion, otherwise LGTM. Thanks for this :)

**Comment by TheNeuralBit** [comment]
> we'd need to re-run a seed job first

Oh sorry I just saw the Run Seed Job but didn't notice it was last night.

**Comment by TheNeuralBit** [comment]
Run XVR_Direct PostCommit

**Comment by tvalentyn** [comment]
Noting that all tests passed.

File: .test-infra/jenkins/README.md (hunk 1 above)
Comment (tvalentyn) [reply .test-infra/jenkins/README.md:112]: Done, thanks.

File: release/src/main/python-release/python_release_automation_utils.sh (hunk 2 above)
Comment (tvalentyn) [reply release/src/main/python-release/python_release_automation_utils.sh:92]: Yes, the format of wheels changed in 3.8. I expect it to continue in 3.9, although haven't seen 3.9 wheels yet.

From https://wheel.readthedocs.io/en/stable/news.html:
```
0.33.5 (2019-08-17)

Don’t add the m ABI flag to wheel names on Python 3.8 (PR by rdb)
Updated MANIFEST.in to include many previously omitted files in the sdist
```

Comment (TheNeuralBit) [reply release/src/main/python-release/python_release_automation_utils.sh:92]: :+1: thanks

File: sdks/python/apache_beam/__init__.py (hunk 3 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/__init__.py:90]: python_requires >= 3.6 should make it uninstallable on <3.5, but for folks like us who import sources and don't `pip install` beam a warning wouldn't hurt.

**Comment by tvalentyn** [comment]
Opened https://issues.apache.org/jira/browse/BEAM-10955 for Java Flink runner flake. Merging based on passing precommit singals and prior green test signals to unblock testing at HEAD.

**Pull request merged**